pkg/agave/validator.go

```diff
@@ -38,9 +82,29 @@ func (cmd *InstallCommand) Env() map[string]string {
 		env["VALIDATOR_VERSION"] = *cmd.Version
 	}
 
+	additionalEnv := cmd.getValidatorEnv()
+	if len(additionalEnv) > 0 {
+		env["VALIDATOR_ENV"] = strings.Join(additionalEnv, " ")
+	}
+
 	return env
 }
 
+func (cmd *InstallCommand) getValidatorEnv() []string {
+	var validatorEnv []string
+
+	if cmd.Metrics != nil {
+		metricsEnv, err := cmd.Metrics.ToEnv()
+		if err != nil {
+			fmt.Printf("Warning: Invalid metrics URL: %v\n", err)
+		} else {
+			validatorEnv = append(validatorEnv, metricsEnv)
+		}
+	}
+
+	return validatorEnv
+}
+
```

Comment:
```golang
func (cmd *InstallCommand) Env() map[string]string {
        env := Environment {
           Metrics: cmd.Metrics
        };
        
	manifest := map[string]string{
		"VALIDATOR_FLAGS":      strings.Join(cmd.Flags.toArgs(), " "),
		"IDENTITY_KEYPAIR":     cmd.KeyPairs.Identity,
		"VOTE_ACCOUNT_KEYPAIR": cmd.KeyPairs.VoteAccount,
		"VALIDATOR_ENVIRONMENT": env.ToString(),
	}
	
	if cmd.Version != nil {
		env["VALIDATOR_VERSION"] = *cmd.Version
	}

	return env
}

// The runtime environment for the agave validator 
type Environment struct {
  metrics: *Metrics,
}

fn (env *Environment) ToString() string {
   gather string := ""
   
   if cmd.Metrics != nil {
     metrics, err := env.Metrics.ToEnv()
		
     if err == nil {
            gather = append(gather, metrics)
     } else {
            fmt.Printf("Warning: Invalid metrics URL: %v\n", err)
     }
   }

    return gather
}
```

What do you think about this? If more runtime environment is needed can keep feeding the Environment struct.